packages/components/src/MultiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { HTMLAttributes, useRef, useId } from "react"
+import classnames from "classnames"
+import { ReactFocusOnProps } from "react-focus-on/dist/es5/types"
+import { Heading } from "~components/Heading"
+import { OverrideClassName } from "~types/OverrideClassName"
+import {
+  MultiSelectOptions,
+  MultiSelectOptionsProps,
+} from "./subcomponents/MultiSelectOptions"
+import { MultiSelectToggle } from "./subcomponents/MultiSelectToggle"
+import { Popover, useFloating } from "./subcomponents/Popover"
+import { MultiSelectOption } from "./types"
+import styles from "./MultiSelect.module.scss"
+
+export type MultiSelectProps = {
+  label: string
+  items: MultiSelectOptionsProps["options"]
+  selectedValues: Set<MultiSelectOption["value"]>
+  onSelectedValuesChange: MultiSelectOptionsProps["onChange"]
+  isOpen: boolean
+  onOpenChange: (isOpen: boolean) => void
+} & OverrideClassName<HTMLAttributes<HTMLDivElement>>
+
+export const MultiSelect = ({
+  id: propsId,
+  label,
+  items,
+  selectedValues,
+  onSelectedValuesChange,
+  isOpen,
+  onOpenChange,
+  classNameOverride,
+  ...restProps
+}: MultiSelectProps): JSX.Element => {
+  const id = propsId ?? useId()
+
+  const toggleButtonRef = useRef<HTMLButtonElement>(null)
+  const { refs } = useFloating()
+
+  const handleToggleClick = (): void => onOpenChange(!isOpen)
+  const handleClose = (): void => onOpenChange(false)
+
+  const onClickOutside: ReactFocusOnProps["onClickOutside"] = e => {
+    const toggle = refs.reference.current as Node
+    const isInToggle = toggle.contains(e.target as HTMLElement)
+    if (!isInToggle) {
+      e.stopPropagation()
+      handleClose()
+    }
+  }
+
+  const itemsMap = items.reduce(
+    (acc, item) => {
+      acc[item.value] = item
+      return acc
+    },
+    {} as Record<MultiSelectOption["value"], MultiSelectOption>
+  )
+
+  return (
+    <div id={id} className={classnames(classNameOverride)} {...restProps}>
+      <Heading tag="span" variant="heading-6" id={`${id}--label`}>
```

Comment:
This combination is pretty weird, maybe flag this with design to consider either changing the label styles or discussing adding this style to the `Text` component...

Reply:
Out of scope :P These are existing components. If you look at the stylesheet for `Label`, you'll find it all references the styles of Heading 6.

Reply:
Yeah, I mean flag it with them so they can add it to their backlog 😄

Reply:
I don't think it's a designer problem. It's more an engineering one. It's probably a coincidence that they look the same, and the implementation wasn't done right. Probably needs discussion with both teams.